Doc/whatsnew/3.14.rst

```diff
@@ -2099,38 +2099,22 @@ typing
 
 .. _whatsnew314-typing-union:
 
-* :class:`types.UnionType` and :class:`typing.Union` are now aliases for each other,
-  meaning that both old-style unions (created with ``Union[int, str]``) and new-style
-  unions (``int | str``) now create instances of the same runtime type. This unifies
+* Both old-style unions (created with ``Union[int, str]``) and new-style
+  unions (``int | str``) now create instances of the same runtime type,
+  :class:`types.UnionType`. This unifies
   the behavior between the two syntaxes, but leads to some differences in behavior that
   may affect users who introspect types at runtime:
 
   - Both syntaxes for creating a union now produce the same string representation in
     ``repr()``. For example, ``repr(Union[int, str])``
     is now ``"int | str"`` instead of ``"typing.Union[int, str]"``.
-  - Unions created using the old syntax are no longer cached. Previously, running
```

Comment:
Not sure why this was removed; the text would seem to still apply under this PR.